api/src/main/java/org/pmiops/workbench/db/model/DbInstitution.java

```diff
@@ -11,88 +12,128 @@
     this.displayName = displayName;
+    return this;
   }
 
-  public Short getOrganizationTypeEnum() {
-    return organizationTypeEnum;
+  @Column(name = "organization_type_enum")
+  public OrganizationType getOrganizationTypeEnum() {
+    return DbStorageEnums.organizationTypeFromStorage(organizationTypeEnum);
   }
 
-  public void setOrganizationTypeEnum(Short organizationTypeEnum) {
-    this.organizationTypeEnum = organizationTypeEnum;
+  public DbInstitution setOrganizationTypeEnum(OrganizationType type) {
+    this.organizationTypeEnum = DbStorageEnums.organizationTypeToStorage(type);
+    return this;
   }
 
+  @Column(name = "organization_type_other_text")
   public String getOrganizationTypeOtherText() {
     return organizationTypeOtherText;
   }
 
-  public void setOrganizationTypeOtherText(String organizationTypeOtherText) {
+  public DbInstitution setOrganizationTypeOtherText(String organizationTypeOtherText) {
     this.organizationTypeOtherText = organizationTypeOtherText;
+    return this;
   }
 
+  @OneToMany(mappedBy = "institution", cascade = CascadeType.ALL, fetch = FetchType.EAGER)
+  @NotNull
   public Set<DbInstitutionEmailDomain> getEmailDomains() {
     return emailDomains;
   }
 
-  public void setEmailDomains(Iterable<DbInstitutionEmailDomain> emailDomains) {
-    this.emailDomains =
-        StreamSupport.stream(emailDomains.spliterator(), false).collect(Collectors.toSet());
+  /**
+   * Effectively: do an in-place this.emailDomains = emailDomains
+   *
+   * <p>Hibernate doesn't like it when you reassign collections. Instead, modify in-place. First,
+   * call retainAll() to subset DB rows to those we wish to keep: the intersection of old and new.
+   * Then, call addAll() to add the diff(new - old) rows.
+   *
+   * <p>https://stackoverflow.com/questions/5587482/hibernate-a-collection-with-cascade-all-delete-orphan-was-no-longer-referenc
+   *
+   * @param emailDomains the new set of domains for this Institution
+   */
+  public DbInstitution setEmailDomains(final Set<DbInstitutionEmailDomain> emailDomains) {
+    final Set<DbInstitutionEmailDomain> attachedDomains =
+        emailDomains.stream()
+            .map(domain -> domain.setInstitution(this))
+            .collect(Collectors.toSet());
+    // modifies this set so that its value is the intersection of the two sets
+    this.emailDomains.retainAll(attachedDomains);
+    this.emailDomains.addAll(Sets.difference(attachedDomains, this.emailDomains));
+    return this;
   }
 
+  @OneToMany(mappedBy = "institution", cascade = CascadeType.ALL, fetch = FetchType.EAGER)
+  @NotNull
   public Set<DbInstitutionEmailAddress> getEmailAddresses() {
     return emailAddresses;
   }
 
-  public void setEmailAddresses(Iterable<DbInstitutionEmailAddress> emailAddresses) {
-    this.emailAddresses =
-        StreamSupport.stream(emailAddresses.spliterator(), false).collect(Collectors.toSet());
+  /**
+   * Effectively: do an in-place this.emailAddresses = emailAddresses
```

Comment:
So the intention of this function is to replace all matching email addresses with updated ones (not sure what other fields that would involve). If this isn't bound directly to the table(s), then I'm confused about the use case.